### PR TITLE
Update VS Publish to Azure

### LINKED ISSUE #26774: Outdated version of VS used in the article
The project creation, the publish and the provisioning all look different in the latest version of VS. Also the version of the ASP.NET template being used is old as well.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 71a44c72-d256-fbee-ea70-2eb7472f8182
* Version Independent ID: 157027ff-0bad-828d-a442-a98acd219fba
* Content: [Publish an ASP.NET Core app to Azure with Visual Studio](https://docs.microsoft.com/en-us/aspnet/core/tutorials/publish-to-azure-webapp-using-vs?view=aspnetcore-6.0)
* Content Source: [aspnetcore/tutorials/publish-to-azure-webapp-using-vs.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/tutorials/publish-to-azure-webapp-using-vs.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-tutorials**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

---
[Associated WorkItem - 33235](https://dev.azure.com/msft-skilling/Content/_workitems/edit/33235)

### PR DESCRIPTION
**Pull request by wadepickett**
[Internal doc build review](https://review.learn.microsoft.com/en-us/aspnet/core/tutorials/publish-to-azure-webapp-using-vs?view=aspnetcore-7.0&branch=pr-en-us-27185)

Update the VS publish to Azure topic.

New instruction, screenshots and code.

Fixes #26774

### CODE REVIEW

**Comment by wadepickett** [comment]
@CamSoper,  Thanks for testing this out earlier.  When running the steps on this new version, I am missing something.  I can't get the EF file publishing options.

image

You were able to get them to show once earlier.  For some reason I can't replicate success and I am missing something.

I did check for gotchas like: 
- Accidently leaving the app running so an EF related build was not blocked, that was not it.  
- I went back and tried a .NET 6.0 version, same issue, not a .NET 7 specific issue.  
- Verified I didn't have a mismatch of the Global EF tool + packages.  That was not it.  
- Verified I didn't have an old DB by the same name still on my local before running the steps.

Maybe it is an issue with what my subscription allows/does not allow?

Since you were able to run through this tutorial OK earlier, could you run through the steps one more time and see if you get the EF setings for the File publish options?  If you do.  Try running it on Azure Web App as it suggests but do the extra step of trying to add an new user or login to see if it is connecting to the EF generated DB, it should reach the login page if that exists.  Thanks so much for your help.

I am leaving this in draft for now until we know there is a successful set of steps here.

**Comment by Rick-Anderson** [comment]
@wadepickett  you probably need to stop Kestrel so it will unlock the JSON file. **Build > Clean**

It's there

image

@CamSoper any reason to use Linux?

**Comment by wadepickett** [comment]
@Rick-Anderson, ..."stop Kestrel so it will unlock the JSON file. Build > Clean"
THANKS!  I'll test again and try that.  I will also add it to the instructions.  Thanks also for taking a screenshot while you were there.

**Comment by wadepickett** [comment]
Linux:  I just updated the whole thing for this Linux target, if we are going to change to windows at the last moment, then I would rather publish what I just finished for Linux and get it out there, then in a separate issue add a tab so we have both windows and Linux represented which would be best I think anyway.  We should represent both when we can.

**Comment by Rick-Anderson** [comment]
> Linux: I just updated the whole thing for this Linux target, if we are going to change to windows at the last moment, then I would rather publish what I just finished for Linux and get it out there, then in a separate issue add a tab so we have both windows and Linux represented which would be best I think anyway. We should represent both when we can.

Create an issue Pri3 and when you close all your higher priority issues you can do two tabs.

**Comment by wadepickett** [comment]
Linux: thanks.

I ran Build > Clean and started from scratch for Azure Web App/Linux.  Still the same issue.  Your screenshot showing it worked doesn't indicate "Zip Deploy" at the top as mine does.  Were you following the review doc?  Maybe there is something you assume or do that I'm not in the actual doc steps?

**Comment by Rick-Anderson** [comment]
> Your screenshot showing it worked doesn't indicate "Zip Deploy" at the top as mine does. 

That's the wrong deployment

File: aspnetcore/tutorials/publish-to-azure-webapp-using-vs.md [hunk 1]
```diff
@@ -64,6 +64,7 @@ The app displays the email used to register the new user and a **Logout** link.
 ![Web application open in Microsoft Edge. The Register link is replaced by the text Hello user@example.com!](publish-to-azure-webapp-using-vs/_static/hello.png)
 
 * Stop the application by either closing the browser or in Visual Studio select **Debug** > **Stop Debugging**.
+* In Visual Studio select **Build** > **Clean Solution** to remove any intermediate and output files, and avoid potential file locks.
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/publish-to-azure-webapp-using-vs.md:67]:
```suggestion
* In Visual Studio select **Build** > **Clean Solution** to  clean project items  and avoid file contention.
```

File: aspnetcore/tutorials/publish-to-azure-webapp-using-vs.md [hunk 2]
```diff
@@ -1,78 +1,70 @@
...
-ms.date: 07/10/2019
+monikerRange: '>= aspnetcore-7.0'
+ms.author: wpickett
+ms.custom: "devx-track-csharp, mvc, engagement-fy23"
+ms.date: 11/16/2022
 uid: tutorials/publish-to-azure-webapp-using-vs
 ---
 # Publish an ASP.NET Core app to Azure with Visual Studio
 
-By [Rick Anderson](https://twitter.com/RickAndMSFT)
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/publish-to-azure-webapp-using-vs.md:12]: revert

**Comment by Rick-Anderson** [comment]
@wadepickett  do you have an ETA on completion?

**Comment by CamSoper** [comment]
@Rick-Anderson Wade had a heck of a time figuring out how to get VS to do EF migrations for Linux deployments. Turns out, VS just doesn't even bother trying because it's a zip deployment behind the scenes (not WebDeploy). 

@wadepickett LMK if you want my help. I'm pretty sure my previous suggestion should work, but I'm happy to help come up with others. 🙂

**Comment by Rick-Anderson** [comment]
> @Rick-Anderson Wade and I had a heck of a time figuring out how to get VS to do EF migrations for Linux deployments. Turns out, VS just doesn't even bother trying because it's a zip deployment behind the scenes (not WebDeploy).
> 
> @wadepickett LMK if you want my help. I'm pretty sure my previous suggestion should work, but I'm happy to help come up with others. 🙂


Let's just use a Windows deployment like we do with all the other tutorials. We'll leave EF Migrations on Linux VMs to the EF folks.

**Comment by Rick-Anderson** [comment]
> @Rick-Anderson Wade and I had a heck of a time figuring out how to get VS to do EF migrations for Linux deployments. Turns out, VS just doesn't even bother trying because it's a zip deployment behind the scenes (not WebDeploy).
> 

@AngelosP @ajcvickers @codemillmatt

Where's the right place to get EF migrations on Azure  Linux deployments documented?

**Comment by Rick-Anderson** [comment]
@wadepickett  do you need me or @tdykstra  to take over? It's been 3.5 months. If not, do you have an ETA?

**Comment by wadepickett** [comment]
@Rick-Anderson, you asked me to stop on this at the time citing low traffic and to hit other higher priority items when it was nearly done.  Then I was on vacation for a month and half and back to pri 0-1 items.  I will get this complete.  Sorry it has been left hanging here so long.

**Comment by wadepickett** [comment]
@Rick-Anderson, I am finishing the 10 tutorial update for the MVC series for Mac you set to Pri0 #27383  this weekend (today) and then I will get my head back in this one and get it done on Monday (Tomorrow).  Again, sorry this sat so long.

**Comment by Rick-Anderson** [comment]
> @Rick-Anderson, you asked me to stop on this at the time citing low traffic and to hit other higher priority items when it was nearly done. 

I never said stop, I just reminded you we had high priority .NET 7 release topics to complete. If it's nearly done, it's best to finish it. My concern was

**Review by tdykstra — APPROVED** [review]
Looks good - I left some minor suggestions.

File: aspnetcore/tutorials/publish-to-azure-webapp-using-vs.md [hunk 3]
```diff
@@ -167,15 +167,15 @@ Click **Publish**. Visual Studio publishes your app to Azure. When the deploymen
 
 ### Update the app
```
[the remark this answers is not in the document]
Comment (tdykstra) [reply aspnetcore/tutorials/publish-to-azure-webapp-using-vs.md:166]: On second thought I think this is more than a nice-to-have -- if you don't register and log in you haven't validated the database deployment and run-time connection.

File: aspnetcore/tutorials/publish-to-azure-webapp-using-vs.md [hunk 4]
```diff
@@ -88,18 +84,18 @@ In the **Publish** dialog:
...
 In the **Publish** dialog:
 
-* Select **Azure App Service (Linux)**.
+* Select **Azure App Service (Windows)**.
 * Select **Next**.
 
 ![Publish Dialog: select Azure Service](publish-to-azure-webapp-using-vs/_static/maas2.png)
 
-In the **Publish** dialog, select **Create a new Azure App Service**.
+In the **Publish** dialog, in the **App Service** pane, select **Create new**.
```
[the remark this answers is not in the document]
Comment (wadepickett) [reply aspnetcore/tutorials/publish-to-azure-webapp-using-vs.md:90]: You are right, it should be tab really.  Fixing....

**Pull request merged**